### PR TITLE
feat: agent shutdown

### PR DESCRIPTION
**Pull request by robot9706**
- Added `SHUTDOWN` close reason to agent: DAgent tries the DockerAPI first, then a simple process exit
- When a node is deleted the shutdown command is invoked

### CODE REVIEW

**Comment by polaroi8d** [comment]
Please add the scope for the PR and fix if it contains only the `dagent` part. Additionally add todo for the crane, please.

File: golang/pkg/dagent/dagent.go [hunk 1]
```diff
@@ -50,6 +51,13 @@ func Serve(cfg *config.Configuration) {
 func grpcClose(ctx context.Context, reason agent.CloseReason) error {
 	if reason == agent.CloseReason_SELF_DESTRUCT {
 		return update.RemoveSelf(ctx)
+	} else if reason == agent.CloseReason_SHUTDOWN {
+		log.Info().Msg("Remote shutdown requested")
+		err := utils.StopSelf(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Docker failed to stop")
+			os.Exit(0)
```
Comment (nandor-magyar) [inline golang/pkg/dagent/dagent.go:59]: Stopping should be done in all environment using Exit 0, is there an issue with it?

Comment (robot9706) [reply golang/pkg/dagent/dagent.go:59]: Docker restarts the container for some reason, but `ContainerStop` keeps it stopped.

Comment (nandor-magyar) [reply golang/pkg/dagent/dagent.go:59]: It is the `unless-stopped` restart policy's behavior, `on-failure` is the one we are looking for if we want 'exit 0' to be a valid termination state for dagent.

Comment (robot9706) [reply golang/pkg/dagent/dagent.go:59]: I see, I'll change the install scripts then, and I'll be able to implement the Kubernetes exit as well I think.

**Review by m8vago — APPROVED** [review]

**Review by nandor-magyar — APPROVED** [review]
LGTM

**Pull request merged**